pkg/sql/schema_change_plan_node.go

```diff
@@ -218,15 +213,29 @@ func (p *planner) waitForDescriptorSchemaChanges(
 			if err != nil {
 				return err
 			}
-			blocked = desc.HasConcurrentSchemaChanges()
+			isBlocked = desc.HasConcurrentSchemaChanges()
 			return nil
 		}); err != nil {
 			return err
 		}
-		if !blocked {
+		if isBlocked {
+			wasBlocked = true
+		} else {
 			break
 		}
+		if logEvery.ShouldLog() {
+			log.Infof(ctx,
+				"schema change waiting for concurrent schema changes on descriptor %d,"+
+					" waited %v so far", descID, timeutil.Since(start),
+			)
+		}
```

Comment:
moving this to the end of the for-loop makes more sense, so that no message is logged if we're not actually waiting